src/server/scripts/Commands/cs_misc.cpp

```diff
@@ -723,8 +723,7 @@ class misc_commandscript : public CommandScript
             // before GM
             float x, y, z;
             handler->GetSession()->GetPlayer()->GetClosePoint(x, y, z, target->GetObjectSize());
-            if (target->TeleportTo(handler->GetSession()->GetPlayer()->GetMapId(), x, y, z, target->GetOrientation(), 0, handler->GetSession()->GetPlayer()))
-                target->SetPhaseMask(handler->GetSession()->GetPlayer()->GetPhaseMask(), false);
+            target->TeleportTo(handler->GetSession()->GetPlayer()->GetMapId(), x, y, z, target->GetOrientation(), 0, handler->GetSession()->GetPlayer());
```

Comment:
Interesting, so someone deliberately chose to copy the summoning player's phase mask. This makes it a lot easier to change.

Reply:
Looking at the git history, this line has been here since the Sunwell code, so I can't find any record of why it was added +/or if there was discussion around it. 

https://github.com/azerothcore/azerothcore-wotlk/blob/e8e94a0a663be0a1cefe704d8c6860a526b2d7e9/src/server/scripts/Commands/cs_misc.cpp#L749
https://github.com/Kittnz/Sunwell/blame/master/src/server/scripts/Commands/cs_misc.cpp#L749

Reply:
Ok so I decided to go down the rabbit hole further + found the initial trinity core commit that added this.
https://github.com/TrinityCore/TrinityCore/commit/35c57db00aaaee504baeaa243640eab5f5172f27

My interpretation of this is that they figured that when teleporting to a player you'd want to inherit their phase, and they decided that it naturally followed that teleporting a player to you should also copy their phase. What they didn't think of was that GMs could see all phases, and you wouldn't want this to be applied to a player a GM was moving around. This does assume that GMs seeing all phases was even a thing back in 2009.

Reply:
@steenburgh test if you can and post result in a general comment, thanks for the investigation.

Reply:
I'll test this later today (US Central time) if I can.